modules/drivers/redshift/src/metabase/driver/redshift.clj

```diff
@@ -504,3 +504,15 @@
 (defmethod driver.sql/default-database-role :redshift
   [_ _]
   "DEFAULT")
+
+(defmethod driver/add-columns! :redshift
+  [driver db-id table-name column-definitions & {:as settings}]
+  ;; Redshift doesn't support adding multiple columns at a time, so we break it up
+  (let [f (get-method driver/add-columns! :postgres)]
+    (doseq [[k v] column-definitions]
+      (f driver db-id table-name {k v} settings))))
+
+(defmethod driver/alter-columns! :redshift
+  [_driver _db-id _table-name _column-definitions]
+  ;; TODO: redshift doesn't allow promotion of ints to floats using ALTER TABLE.
+  (throw (ex-info "Uploaded value with the wrong type" {})))
```

Comment:
Controversial but I'm lazy to fix this the proper way. See PR description for an explanation

Reply:
I'm OK with this failing but it would be really nice to have a better failure message - at least the column and ideally the first "relaxing" value.

Given this isn't in a tight-loop, a try-catch is probably fine although I always shudder at exceptions for control flow.

Reply:
Adding the column and type to the error message a good idea, since we have them, but how do you imagine getting the first "relaxing" value in the error message? That would have to be in the `upload.types` code, and I was trying to avoid polluting that with driver-specific stuff.

Reply:
super easy: "Uploaded a value with type Y in column X"
not-so easy: "Uploaded value Z with type Y in column X"

the latter is nicer obviously but we can't do it from driver code currently

Reply:
updated:
https://github.com/metabase/metabase/blob/1f66af770e3330ecb157fc3c367fc642035851fc/test/metabase/upload_test.clj#L1627

good enough to ship imo

Reply:
Yeah it would have to be a horrible hack unless we rework things, this is great for now.